docs/sql-migration-guide.md

```diff
@@ -28,6 +28,8 @@ license: |
 
   - In Spark 3.1, SQL UI data adopts the `formatted` mode for the query plan explain results. To restore the behavior before Spark 3.0, you can set `spark.sql.ui.explainMode` to `extended`.
 
+  - In Spark 3.1, the column metadata will always be propagated in the API `name` and `as`. In Spark version 3.0 and earlier, the metadata of `NamedExpression` is set as the `explicitMetadata` for the new column. To restore the behavior before Spark 3.0, you can use the API `as(alias: String, metadata: Metadata)` with explicit metadata.
```

Comment:
I think the patch has been merged to 3.0?

Reply:
in the API `name` and `as` -> in the API `Column#name` and `Column#as`

Reply:
> In Spark version 3.0 and earlier, the metadata of `NamedExpression` is set as the `explicitMetadata` for the new column.

I think the old behavior is using the metadata of `NamedExpression` at the time the API was called. The metadata won't change even if the underlying `NamedExpression` changes metadata.

Reply:
Thanks, let me emphasize the usage of explicitMetadata.